Detectors/Upgrades/ALICE3/FCT/simulation/src/Detector.cxx

```diff
@@ -0,0 +1,592 @@
+    } else if (layerType == 2) {
+      auto& thisLayer = mConverterLayers.emplace_back(0, layerNumber, layerName, z_layer, r_in, r_out_l_side, Layerx2X0, layerType);
+    }
+    layerNumber++;
+  }
+
+  mNumberOfLayers = layerNumber;
+  LOG(info) << " Loaded FCT Detector with  " << mNumberOfLayers << " layers";
+  LOG(info) << " Of which " << layerNumberDisk << " are disks";
+  LOG(info) << " Of which " << layerNumberSquare << " are disks";
+}
+
+//_________________________________________________________________________________________________
+void Detector::exportLayout()
+{
+  // Export FCT Layout description to file. One line per disk
+  // z_layer r_in r_out Layerx2X0
+
+  TDatime* time = new TDatime();
+  TString configFileName = "FCT_layout_";
+  configFileName.Append(Form("%d.cfg", time->GetTime()));
+
+  // std::string configFileName = "FCT_layout.cfg";
+
+  LOG(info) << "Exporting FCT Detector layout to " << configFileName;
+
+  std::ofstream fOut(configFileName, std::ios::out);
+  if (!fOut) {
+    printf("Cannot open file\n");
+    return;
+  }
+  fOut << "#   z_layer   r_in   r_out_l_side   Layerx2X0" << std::endl;
+  for (auto layer : mBackwardLayers) {
+    if (layer.getType() == 0) {
+      fOut << layer.getZ() << "  " << layer.getInnerRadius() << "  " << layer.getOuterRadius() << "  " << layer.getx2X0() << std::endl;
+    } else if (layer.getType() == 1) {
+      fOut << layer.getZ() << "  " << layer.getInnerRadius() << "  " << layer.getSideLength() << "  " << layer.getx2X0() << std::endl;
+    }
+  }
+}
+
+//_________________________________________________________________________________________________
+void Detector::buildBasicFCT(const FCTBaseParam& param)
+{
+  // Build a basic parametrized FCT detector with nLayers equally spaced between z_first and z_first+z_length
+  // Covering pseudo rapidity [etaIn,etaOut]. Silicon thinkness computed to match layer x/X0
+
+  LOG(info) << "Building FCT Detector: Conical Telescope";
+
+  auto z_first = param.z0;
+  auto z_length = param.zLength;
+  auto etaIn = param.etaIn;
+  auto etaOut = param.etaOut;
+  auto Layerx2X0 = param.Layerx2X0;
+  mNumberOfLayers = param.nLayers;
+  mLayerID.clear();
+
+  Int_t type = 0; // Disk
+
+  for (Int_t layerNumber = 0; layerNumber < mNumberOfLayers; layerNumber++) {
+    std::string layerName = GeometryTGeo::getFCTLayerPattern() + std::to_string(layerNumber); // + mNumberOfLayers * direction);
+    mLayerName.push_back(layerName);
+
+    // Adds evenly spaced layers
+    Float_t layerZ = z_first + (layerNumber * z_length / (mNumberOfLayers - 1)) * std::copysign(1, z_first);
+    Float_t rIn = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaIn))));
+    Float_t rOut = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaOut))));
+    auto& thisLayer = mBackwardLayers.emplace_back(0, layerNumber, layerName, layerZ, rIn, rOut, Layerx2X0, type);
```

Comment:
you are not using `thisLayer`.
Only `mBackwardLayers.emplace_back(0, layerNumber, layerName, layerZ, rIn, rOut, Layerx2X0, type);` will do

Reply:
Done